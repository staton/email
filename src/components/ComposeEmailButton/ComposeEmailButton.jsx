import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    content: PropTypes.any.isRequired,
    onClick: PropTypes.func,
    title: PropTypes.string
};

const defaultProps = {
    onClick: () => {},
    title: ''
};

export class ComposeEmailButton extends Component {

    constructor(props) {
        super(props);
    }

	render() {
		return (
            <div className="ComposeEmailButton">
                <button 
                    className="ComposeEmailButton__button"
                    onClick={this.props.onClick}
                    title={this.props.title}
                >
                    {this.props.content}
                </button>
            </div>
		);
    }

}

ComposeEmailButton.propTypes = propTypes;
ComposeEmailButton.defaultProps = defaultProps;

export default ComposeEmailButton;