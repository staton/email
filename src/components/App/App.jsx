import React, { Component } from 'react';
import ComposeEmailButton from '../ComposeEmailButton/ComposeEmailButton';
import Drawer from '../Drawer/Drawer';
import EmailList from '../EmailList/EmailList';
import Header from '../Header/Header';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
import LoadingOverlayState from '../../enums/LoadingOverlay';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { MdEdit } from 'react-icons/md';
import { setScreenSize } from '../../redux/actions/app-actions';
import { 
    loadEmails,
    setInboxListItemsActive 
} from '../../redux/actions/email-actions';

export class App extends Component {

    constructor() {
        super();

        this.handleResized = this.handleResized.bind(this);
    }

    componentDidMount() {
        // listen for window resize events:
        window.addEventListener('resize', this.handleResized);

        // load the initial emails:
        this.props.loadEmails();
    }

	render() {
		return (
			<div className="App">
				<Header />
                <div className="App__content">
                    <Drawer />
                    {
                        (this.props.emailLoadingOverlayState === LoadingOverlayState.None)
                        ? null
                        : <LoadingOverlay loadingOverlayState={this.props.emailLoadingOverlayState} />
                    }
                    <EmailList 
                        isListActive={this.props.isInboxListActive}
                        setListActive={this.props.setInboxListItemsActive}
                    />
                </div>
                {
                    // For small screens, the New Email button will be a FAB.
                    // For larger screens, it will be in the drawer.
                    (this.props.isSmallScreen)
                    ?   <ComposeEmailButton icon={<MdEdit />} />
                    :   null
                }
			</div>
		);
    }

    /**
     * Called when the window resizes.
     * @param {object} e The event.
     */
    handleResized(e) {
        this.props.setScreenSize(window.innerWidth, window.innerHeight);
    }

}

function mapStateToProps(store, ownProps) {
    return {
        emailLoadingOverlayState: store.email.loadingOverlayState,
        isInboxListActive: store.email.isInboxListActive,
        isSmallScreen: store.app.isSmallScreen
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        loadEmails: loadEmails,
        setScreenSize: setScreenSize,
        setInboxListItemsActive: setInboxListItemsActive
    },
    dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(App);