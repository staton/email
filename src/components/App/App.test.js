import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ComposeEmailButton from '../ComposeEmailButton/ComposeEmailButton';
import { App } from './App';

describe('<App />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<App loadEmails={jest.fn()} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('contains a ComposeEmailButton component for small screens', () => {
        const wrapper = shallow(
            <App
                isSmallScreen={true}
                loadEmails={jest.fn()}
            />
        );
        expect(wrapper.find(ComposeEmailButton).length).toBe(1);
    });

    it('does not contain a ComposeEmailButton component for large screens', () => {
        const wrapper = shallow(
            <App
                isSmallScreen={false}
                loadEmails={jest.fn()}
            />
        );
        expect(wrapper.find(ComposeEmailButton).length).toBe(0);
    });

});


