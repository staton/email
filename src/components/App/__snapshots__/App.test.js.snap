// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <div
    className="App__content"
  >
    <Connect(Drawer) />
    <Connect(EmailList) />
  </div>
</div>
`;
